### ISSUE TITLE
ClientRequest patch is never restored

### ISSUE THREAD

**Issue by kettanaito** [mswjs/interceptors#115]
The following patch of the `ClientRequest` class doesn't seem to get restored anywhere:

https://github.com/mswjs/interceptors/blob/16fd35622efe6328497ade15e21a84f6afda0a93/src/interceptors/ClientRequest/index.ts#L47

## Solution

Restore the `http.ClientRequest` to the `pureClientRequest` (stored original class) in the cleanup function of `interceptClientRequest`:

https://github.com/mswjs/interceptors/blob/16fd35622efe6328497ade15e21a84f6afda0a93/src/interceptors/ClientRequest/index.ts#L113-L122

## GitHub

- Originated from https://github.com/mswjs/msw/issues/710

**Comment by mathanpec** [comment]
Hi @kettanaito,

I see a small problem with just restoring `http.ClientRequest` with `pureClientRequest` since the `pureClientRequest` is getting initialised itself only when the actual request gets fired inside `requestOverride`.  But in large code based with multiple test files `server.listen()` and `server.close()` will happen for every test file irrespective of whether the request is going to be fired inside a test file or not. So in that case `pureClientRequest` will not be initialised and when we do `server.close()` after the test file execution we will end up setting `undefined` to `http.ClientRequest`. We can have null check there but that kinda feels hacky and it will be difficult to recall later as to why that null condition was added.

So on further looking into the code, I feel we can completely remove the monkey patching of `http.ClientRequest` itself and directly return object created out of `ClientRequestOverride` itself from function `handleClientRequest`. Since `ClientRequestOverride` is already inheriting from 'http.ClientRequest', it will have access to all the proto functions of `http.ClientRequest` also.

After removing the monkey patching completely, I ran the test cases and they all look green. Also heap memory also stays constant between the test files after this.
